### PR TITLE
Add Authentication & Authorization

### PR DESCRIPTION
**Pull request by hsluoyz**
https://github.com/casbin/awesome-auth

awesome auth is a curated list of authentication & authorization software and libraries . And this list exists because it benefits many groups of people — system administrators, security researchers and the developers who care about the security of their systems.

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).** Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. I have reviewed: https://github.com/sindresorhus/awesome/pull/1366 and https://github.com/sindresorhus/awesome/pull/1390.
- I have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title in the format `Add Name of List` (Example: `Add Swift`), not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should:
	- Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- Be added at the bottom of the appropriate category.
- The list I'm submitting complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- It's the result of hard work and the best I could possibly produce.
- Non-generated Markdown file in a GitHub repo.
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Includes a project logo/illustration whenever possible.
	- Either fullwidth or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- The image should be high-DPI. Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of sub-lists, preferably none.
- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
	- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
	- [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
	- If you use a license badge, it should be SVG, not PNG.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.

Go to the top and read it again.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -431,6 +431,7 @@
 - [Vehicle Security and Car Hacking](https://github.com/jaredthecoder/awesome-vehicle-security#readme)
 - [Web Security](https://github.com/qazbnm456/awesome-web-security#readme) - Security of web apps & services.
 - [Lockpicking](https://github.com/meitar/awesome-lockpicking#readme) - The art of unlocking a lock by manipulating its components without the key.
+- [Authentication & Authorization](https://github.com/casbin/awesome-auth) - Processes of verifying who you are and that you have access to something.
```
Comment (meitar) [inline readme.md]: Please link directly to the `#readme` section. I.e., change: `https://github.com/casbin/awesome-auth` to `https://github.com/casbin/awesome-auth#readme`

Comment (hsluoyz) [reply readme.md]: @meitar Fixed.

Comment (hsluoyz) [reply readme.md]: Please approve it, thanks!

**Comment by hsluoyz** [comment]
@sindresorhus this PR originates from: https://github.com/sindresorhus/awesome/pull/1042. (I have to create this new one because the previous repo is missing). I have improved the PR based on the suggestions from you and other two reviewers. What do you think about the current state of the PR? Thanks.

**Comment by hsluoyz** [comment]
@erictleung Thanks. I have changed the description to: ``Software libraries of verifying who you are and that you have access to something.`` It is modified from your previous advice here: https://github.com/sindresorhus/awesome/pull/1042#pullrequestreview-141324985

**Comment by hsluoyz** [comment]
@sindresorhus hi, I think I have resolved all current review questions. Can you have a look at this PR?

**Comment by hsluoyz** [comment]
@BasixKOR Thanks for the reviewing. Can you point me out where are the ``List item description must start with valid casing`` errors?

Line 74 is:
```
- [DotNetOpenAuth](https://github.com/DotNetOpenAuth/DotNetOpenAuth) - C# implementation of the OpenID, OAuth protocols.
```

Line 95 is:
```
- [AT&T XACML](https://github.com/att/XACML) - AT&T XACML 3.0 implementation.
```

**Comment by BasixKOR** [comment]
@hsluoyz I think there's no real problem there, you can ignore them.

**Comment by hsluoyz** [comment]
Hi @sindresorhus , I have fixed all 5 reviewers' suggestions. Please review it :)

File: readme.md [hunk 2]
```diff
@@ -431,6 +431,7 @@
 - [Vehicle Security and Car Hacking](https://github.com/jaredthecoder/awesome-vehicle-security#readme)
 - [Web Security](https://github.com/qazbnm456/awesome-web-security#readme) - Security of web apps & services.
 - [Lockpicking](https://github.com/meitar/awesome-lockpicking#readme) - The art of unlocking a lock by manipulating its components without the key.
+- [Authentication & Authorization](https://github.com/casbin/awesome-auth#readme) - Software libraries of verifying who you are and that you have access to something.
```
Comment (sindresorhus) [inline readme.md]: > The entry in the Awesome list should:
> * Include a short description about the project/theme of the list. **It should not describe the list itself.**

**Comment by sindresorhus** [comment]
The heading of your list should be `Authentication & Authorization` for consistency.

**Comment by sindresorhus** [comment]
Drop the `Inspired by the lists awesome, awesome-awesomeness, and awesome-nodejs.` It's not needed.

**Comment by sindresorhus** [comment]
```
❯ awesome-lint https://github.com/casbin/awesome-auth
✖ Linting

  README.md:75:69
  ✖  75:69  List item description must start with valid casing  remark-lint:awesome/list-item
  ✖  96:45  List item description must start with valid casing  remark-lint:awesome/list-item

  2 errors
```

https://github.com/sindresorhus/awesome-lint

**Comment by sindresorhus** [comment]
I would like to see more than just libraries. Include some blog posts, tutorials, links to community resources (Stack Overflow tag, IRC, chat, forums, communities, Reddit, etc. Example: https://github.com/sindresorhus/awesome-electron#community), videos of talks, book, learning material, etc.

**Comment by sindresorhus** [comment]
`Authentication (aka AuthN) and authorization (aka AuthZ) are both security measures.` should be linkified with:

https://en.wikipedia.org/wiki/Authorization
https://en.wikipedia.org/wiki/Authentication

**Comment by sindresorhus** [comment]
Closing for lack of response.

**Pull request closed without merging**